flang/lib/Lower/CharacterExpr.cpp

```diff
@@ -77,7 +77,7 @@ getCompileTimeLength(const fir::CharBoxValue &box) {
 
 /// Detect the precondition that the value `str` does not reside in memory. Such
 /// values will have a type `!fir.array<...x!fir.char<N>>` or `!fir.char<N>`.
-static bool needToMaterialize(mlir::Value str) {
+LLVM_ATTRIBUTE_UNUSED static bool needToMaterialize(mlir::Value str) {
```

Comment:
Can we simply delete this?

Reply:
It's is used in an assert, but we get a warning in Release mode. (LLVM_ATTRIBUTE_UNUSED is misleading, but I did not used [[maybe_unused]] because I think LLVM/mlir does not use it yet).